Update python3-aiohttp_socks to 0.5.5 and add dependencies

ar-jan opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [x ] This ~~is~~ includes ~~a~~ new package**s** and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Overview
- Electrum 4 depends on a newer version of `aiohttp_socks` than is currently packaged ([electrum-4.0.9 update here](https://github.com/void-linux/void-packages/pull/28305)).
- `aiohttp_socks` >= 0.5.4 depends on `python-socks`, added as new package here.
- `python-socks` has optional dependencies on `python-curio` and `python-trio`; however these libraries are needed in order to pass `python-socks` tests, so they are also added as new packages here.
- `python-trio` depends on `sniffio` and `outcome` libraries, also added as new packages here.
- `python-trio` tests also depend on `pyOpenSSL`, `trustme`, and `astor` which are not available, so its tests are skipped for now.

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

Dependencies added:

    python3-socks-1.2.0
    python3-curio-1.4
    python3-trio-0.18.0
    python3-outcome-1.1.0
    python3-sniffio-1.2.0

sgn commented on srcpkgs/python3-python-socks/template:
```diff
@@ -0,0 +1,20 @@
+# Template file for 'python3-python-socks'
+pkgname=python3-python-socks
```
I prefer to name it `python3-socks`. However, I prefer to include this package when it's actually need it.

ar-jan commented:
@sgn python-socks is needed for python3-aiohttp_socks >= 0.5.4. Electrum 4.x requires aiohttp_socks >= 0.3 but currently 0.2.2 is available in void-packages. So upgrading to latest aiohttp_socks seemed best.

Or should I try to combine all of these in one pull request? (adding python-socks and dependencies, upgrading aiohttp_socks, and upgrading electrum). Doing this in three steps seemed simpler.

ar-jan commented:
I've renamed to python3-socks, and included python3-aiohttp_socks update in this pull request.

Update for Electrum is here #28305 because there's a question about which version of secp256k1 to use as dependency.

Chocimier commented on srcpkgs/python3-aiohttp_socks/template:
```diff
@@ -1,15 +1,15 @@
-pycompile_module="aiohttp_socks"
 hostmakedepends="python3-setuptools"
-depends="python3-aiohttp"
+depends="python3-aiohttp python3-socks>=1.0.1"
```
There was never unmatching version in tree, please remove version spec.

Chocimier commented on srcpkgs/python3-curio/template:
```diff
@@ -0,0 +1,21 @@
+checksum=57edce81c837f3c2cf42fbb346dee26e537d1659e6605269fb13bd179e068744
+
+do_check() {
+	:
```
These packages pass tests with python3-pytest in checkdepends and do_check defined as `PYTHONPATH=$(cd build/lib* && pwd) pytest`, please update.

Chocimier reviewed (commented):
Every python package should depend on either other module or interpreter.
Some packages depend on other modules, eg. trio on attrs.

ar-jan commented:
I've updated the the status and explanation about the dependencies to the issues:

- Electrum 4 depends on a newer version of `aiohttp_socks` than is currently packaged ([electrum-4.0.9 update here](https://github.com/void-linux/void-packages/pull/28305)).
- `aiohttp_socks` >= 0.5.4 depends on `python-socks`, added as new package here.
- `python-socks` has optional dependencies on `python-curio` and `python-trio`; however these libraries are needed in order to pass `python-socks` tests, so they are also added as new packages here.
- `python-trio` depends on `sniffio` and `outcome` libraries, also added as new packages here.
- `python-trio` tests also depend on `pyOpenSSL`, `trustme`, and `astor` which are not available, so its tests are skipped for now.

ar-jan commented:
I can build all included new packages and the updated package locally with `xbps-src pkg -Q`, but for some reason the CI keeps failing on some arches. Any clues?

ar-jan commented:
Closing, updated #28305 with these dependencies.
